parler_tts/modeling_parler_tts.py

```diff
@@ -1211,32 +1867,24 @@ def prepare_inputs_for_generation(
         # apply the delay pattern mask
         input_ids = self.apply_delay_pattern_mask(input_ids, delay_pattern_mask)
 
-        if guidance_scale is not None and guidance_scale > 1:
-            # for classifier free guidance we need to replicate the decoder args across the batch dim (we'll split these
-            # before sampling)
-            input_ids = input_ids.repeat((2, 1))
-            if attention_mask is not None:
-                attention_mask = attention_mask.repeat((2, 1))
-
-            if prompt_hidden_states is not None:
-                prompt_hidden_states = torch.concatenate(
-                    [prompt_hidden_states, torch.zeros_like(prompt_hidden_states)], dim=0
-                )
-
-            if prompt_attention_mask is not None:
-                prompt_attention_mask = torch.concatenate(
-                    [prompt_attention_mask, torch.zeros_like(prompt_attention_mask)], dim=0
-                )
+        position_ids = kwargs.get("position_ids", None)
+        if attention_mask is not None and position_ids is None:
+            # create position_ids on the fly for batch generation
+            position_ids = attention_mask.long().cumsum(-1) - 1
+            position_ids.masked_fill_(attention_mask == 0, 1)
 
         if past_key_values is not None:
             input_ids = input_ids[:, -1:]
+            if position_ids is not None:
+                position_ids = position_ids[:, -input_ids.shape[1]:]
 
             # we only want to use prompt signal in the 1st generation step but keeping the attention mask
             prompt_hidden_states = None
 
         return {
-            "input_ids": input_ids,
+            "input_ids": input_ids.contiguous(),
```

Comment:
Good catch - this is written more explicitly in the llama modelling code now: https://github.com/huggingface/transformers/blob/c85510f958e6955d88ea1bafb4f320074bfbd0c1/src/transformers/models/llama/modeling_llama.py#L1218

Reply:
```suggestion
            "input_ids": input_ids.contiguous(), # `contiguous()` needed for compilation use cases
```